### PR TITLE
Add support for scope related queries

### PR DESCRIPTION
**Pull request by aLagoG**
Added  `get_audio_device_ids_for_scope` which returns devices that are listed under a specific scope instead of just global and `get_audio_device_supports_scope` which checks if a specific devices has any channels for a scope.

This is basically a cleanup of https://github.com/RustAudio/coreaudio-rs/pull/94

### CODE REVIEW

**Review by simlay — APPROVED** [review]
Sorry for the (very) late review. As a person with merge rights to a repo, I should be watching it on github.

I think this is pretty good. It's tough to tell from the github view but I think this is a non-breaking change. I'll test it out locally and merge it.

**Comment by zertosh** [comment]
Thanks @simlay! Can I bug you for a release with this please?

**Comment by simlay** [comment]
> Thanks @simlay! Can I bug you for a release with this please?

https://github.com/RustAudio/coreaudio-rs/pull/105 just waiting for CI to pass.

**Pull request merged**
